### DISCUSSION TITLE
Streamlit custom Component for logout

### DISCUSSION THREAD

**Discussion by MPerrymanAW** [keycloak/keycloak#17407] [Q&A]
Hi, I'm trying to build a Python UI using Streamlit which uses Keycloak for authentication.
I've started using the streamlit-keycloak v1.0.1 package for logging in which is great, but logging off isn't (yet) possible.
I started looking into roughly replicating what was done in the streamlit-keycloak package and can see that its creating a Custom Streamlit Component using the Svelte template as apposed to the React template Streamlit have provided. Unfortunately the source code for the Svelte Component isn't included in the Git Repository so I can't see how it's been done (and it may be using an older version of packages).
I've tried to follow the official Custom Component guide [here](https://docs.streamlit.io/library/components/create) and have imported the keycloak.js file to make use of it's features as per [here](https://www.keycloak.org/docs/latest/securing_apps/index.html#_javascript_adapter) but it always fails when calling the ```init``` function.

I have Keycloak v20.0.0 running in a Production Environment that uses Postgres 15.2 in Docker by following [this](https://www.keycloak.org/server/containers) guide. I've updated the Environment variables of course and I'm still using the certificate generation method specified.
The keycloak.js version I downloaded is also v20.0.0

I couldn't add the example HTML code for the Javascript adapter into the React's render Function's return so I've created a function in the Component which is called in the render Function, examples I've found online suggested adding the ```async``` & ```await``` keywords
```
  private initKeycloak = async () => {
    //you can hardcode these values for now just to see if everything works
    const config = { url: 'https://localhost:8443', realm: 'master', clientId: 'account-console' };

    const keycloak = new Keycloak(config);

    await keycloak
            .init({ onLoad: 'check-sso' })
            .then(isAuthenticated => {
              alert('Authenticated?')
             })
            .catch(error => {
              alert(`keycloak failed to initialise\n${error}`); 
            });
  }
```
I don't understand why the ```init``` is failing, have I potentially missed a setting in Keycloak?

**Comment by codespearhead** [comment]
There's no official Keycloak OIDC client for Python: those [1] are just recommendations, and the same applies to the framework integration you're using. This answer [2][3] goes into more detail about why that’s the case, and [3] in particular might interest you.

Since you're experiencing issues with a third-party component, please open an issue in their respective repository.

Regarding this discussion, could you close it [4] in favor of [2][3]?

***

[1] https://github.com/keycloak/keycloak/blob/98a6ddead0d7121470f5d17be635e1f4dbf3058e/docs/guides/securing-apps/partials/overview/getting-started.adoc?plain=1#L7
[2] https://github.com/keycloak/keycloak/discussions/13510
[3] https://github.com/keycloak/keycloak/discussions/13510#discussioncomment-10925175
[4] https://docs.github.com/en/discussions/managing-discussions-for-your-community/managing-discussions#closing-a-discussion
